compiler/core/js_exp_make.ml

```diff
@@ -1128,6 +1128,31 @@ let or_ ?comment (e1 : t) (e2 : t) =
     | Some e -> e
     | None -> {expression_desc = Bin (Or, e1, e2); comment})
 
+let in_ (prop : t) (obj : t) : t =
+  {expression_desc = In (prop, obj); comment = None}
+
+let has (obj : t) (prop : t) : t =
+  let non_prototype_prop =
+    match prop.expression_desc with
+    | Str
+        {
+          txt =
+            ( "__proto__" | "toString" | "toLocaleString" | "valueOf"
+            | "hasOwnProperty" | "isPrototypeOf" | "propertyIsEnumerable" );
```

Comment:
Looks good to me.

Record semantics are close to `Object.create(null)` and it was actually used in the output of the earlier versions. As we're using object literals for compactness now, this would be a minimal safe guard for the codes with external types.

My previous opinion was that it is not general and safe enough to introduce as a syntax. Having an unsafe primitive for emitting `in` is perfectly. It might be better to prefix with "unsafe"

Reply:
`constructor` too? / `__proto__` is so deprecated, no need to cover